ERCS/erc-7831.md

```diff
@@ -0,0 +1,367 @@
+
+## Motivation
+
+The Ethereum ecosystem is becoming steadily more fragmented. This means a 20 byte address by itself is not enough information to fully specify an account. This can be problematic if funds are sent to an unreachable address on the incorrect chain.
+
+Instead of using chain identifiers, which are not human readable, the address could be extended with a human-readable chain name, which can then be resolved to a chain identifier. The mapping from chain names to identifiers has, since [EIP-155], been maintained off chain using a centralized list. This solution has two main shortcomings:
+
+ - It does not scale with the growing number of L2s.
+ - The list maintainer is a trusted centralized entity.
+
+This ERC proposes the use of ENS to map chain names to identifiers, while still allowing maximum flexibility by changing the root chain.
+
+### Why not ENS with [ERC-2304]?
+
+While [ERC-2304] allows registrants to specify per-chain addresses, it does not provide a default chain to receive assets on (nor should it.) The choice of receiving chain depends too much on off-chain factors to require a transaction to change.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119] and [RFC 8174].
+
+Grammar snippets in this proposal are given in Augmented Backus-Naur form (ABNF) as defined in [RFC 5234] and [RFC 7405].
+
+### Definitions
+
+The following terms are used throughout this proposal:
+
+ - **agent** - software/tool responsible for resolving a chain-specific address to its exact account and chain.
+ - **bridge** - contract that connects the root chain to the target chain (eg. to transfer tokens, proxy function calls.)
+ - **root chain** - blockchain containing bridge and name resolver contracts.
+ - **target chain** - blockchain where the identified account intends to transact; can be any chain with a bridge on the root chain.
+
+### Syntax
+
+At a high level, a chain-specific address is made of three components separated by colons (`:`) ordered from most specific on the left to most general on the right:
+
+ - a `local-part`, that identifies the account on the target chain;
+ - a `chain-part`, that identifies the target chain; and
+ - _optionally_, a `root-part` that identifies the root chain.
+
+These components may be enclosed by parentheses (`(` and `)`) to resolve parsing ambiguities.
+
+More formally, valid chain-specific addresses MUST adhere to the following grammar:
+
+```abnf
+address         = OPEN bare-address CLOSE /
+                  bare-address
+
+bare-address    = local-part SEP chain-part [SEP root-part]
+
+OPEN            = '('
```

Comment:
just a nit:

would prefer if you just use `(` ... etc directly in grammar to have a shorter specification and bundle all the grammar rules together for various parts

Reply:
Like instead of:

> ```abnf
> address         = OPEN bare-address CLOSE /
>                  bare-address
> ```

Use:

```abnf
address         = '(' bare-address ')' /
                  bare-address
```